src/bootstrap/dist.rs

```diff
@@ -398,12 +398,12 @@ impl Step for Rustc {
             // component for now.
             maybe_install_llvm_runtime(builder, host, image);
 
-            let src_dir = builder.sysroot_libdir(compiler, host).parent().unwrap().join("bin");
-            let dst_dir = image.join("lib/rustlib").join(&*host.triple).join("bin");
-            t!(fs::create_dir_all(&dst_dir));
-
             // Copy over lld if it's there
             if builder.config.lld_enabled {
+                let src_dir = builder.sysroot_libdir(compiler, host).parent().unwrap().join("bin");
+                let dst_dir = image.join("lib/rustlib").join(&*host.triple).join("bin");
+                t!(fs::create_dir_all(&dst_dir));
```

Comment:
Other things may end up in the bin dir too in the future, so maybe keep unconditionally creating it?

Reply:
Fixed!